### PR TITLE
Add secure headers

### PR DESCRIPTION
**Pull request by adrazhi**
This PR introduces a small change to GAE http headers by adding the security-relevant http headers. (see below the current analysis of HTTP headers in https://app.terra.bio, in particular the missing headers).

For more info, you may find this [reference](https://infosec.mozilla.org/guidelines/web_security) useful

```
https://app.terra.bio    x-xss-protection           missing_header   This header sets the configuration for the cross-site scripting filter built into most browsers. The recommended value is "X-XSS-Protection: 1; mode=block
https://app.terra.bio    x-content-type-options     missing_header   This header stops a browser from trying to MIME-sniff the content type and forces it to stick with the declared content-type. The only valid value for this header is "X-Content-Type-Options: nosniff".
https://app.terra.bio    feature-policy             missing_header   It is a new header that allows a site to control which features and APIs can be used in the browser.
https://app.terra.bio    referrer-policy            missing_header   It is a new header that allows a site to control how much information the browser includes with navigations away from a document and should be set by all sites.
https://app.terra.bio    content-security-policy    missing_header   This header protects against XSS attacks. By whitelisting sources of approved content, the browser does not load malicious assets.
https://app.terra.bio    strict-transport-security  missing_header   This header strengthens your implementation of TLS by getting the User Agent to enforce the use of HTTPS. The recommended value us "strict-transport-security: max-age=31536000; includeSubDomains".
https://app.terra.bio    x-frame-options            missing_header   This header tells the browser whether the site can be framed. Not allowing framing defends against clickjacking attacks.
https://app.terra.bio    server                     info_disclosure  Google Frontend  server header present. This header gives an attacker info for more targeted attacks.
https://app.terra.bio    content-length             info_header      10634
https://app.terra.bio    content-encoding           info_header      gzip
https://app.terra.bio    x-cloud-trace-context      info_header      0d26585872684dab9e0776d430e8da72;o=1
https://app.terra.bio    age                        info_header      93
https://app.terra.bio    expires                    info_header      Wed, 13 Nov 2019 22:02:08 GMT
https://app.terra.bio    etag                       info_header      "hEFRCA"
https://app.terra.bio    cache-control              info_header      public, max-age=120
https://app.terra.bio    date                       info_header      Wed, 13 Nov 2019 22:00:08 GMT
https://app.terra.bio    content-type               info_header      text/html

```

### CODE REVIEW

File: app.yaml [hunk 1]
```diff
@@ -10,3 +10,8 @@ handlers:
   static_files: build/index.html
   upload: build/index.html
   secure: always
+  http_headers:
+    X-Frame-Options: "DENY"
```
Comment (zarsky-broad) [inline app.yaml:14]: Wouldn't we also want to add `Content-Security-Policy: frame-ancestors 'none'`?
https://infosec.mozilla.org/guidelines/web_security#x-frame-options

File: app.yaml [hunk 2]
```diff
@@ -10,3 +10,8 @@ handlers:
...
+    Strict-Transport-Security: "max-age=2592000; includeSubdomains"
```
Comment (zarsky-broad) [inline app.yaml:15]: Should we look into [HSTS preloading](https://hstspreload.org/?domain=terra.bio)?

File: app.yaml [hunk 3]
```diff
@@ -10,3 +10,8 @@ handlers:
...
+    X-Content-Type-Options: "nosniff"
```
Comment (zarsky-broad) [inline app.yaml:16]: I presume google's automatic MIME types are unaffected by this, yes?

File: app.yaml [hunk 4]
```diff
@@ -10,3 +10,8 @@ handlers:
...
+    X-XSS-Protection: "1; mode=block"
```
Comment (zarsky-broad) [inline app.yaml]: Anything else we should have in [`Content-Security-Policy`](https://infosec.mozilla.org/guidelines/web_security#content-security-policy)?

Nit: Also, end with a newline, plz.

**Review by zarsky-broad — COMMENTED** [review]
Some questions

File: app.yaml (hunk 1 above)
Comment (adrazhi) [reply app.yaml:14]: `X-Frame-Options: deny` takes care of what `Content-Security-Policy: frame-ancestors 'none'` helps with. Also we must look into setting up a more robust CSP, and I'll follow up on that as I want to make sure we don't break anything.

File: app.yaml (hunk 4 above)
Comment (adrazhi) [reply app.yaml]: 👍

File: app.yaml (hunk 2 above)
Comment (adrazhi) [reply app.yaml:15]: It's a good thing to setup for, again does require some more testing since it comes with some "dangerous features" as well.

**Review by zarsky-broad — APPROVED** [review]
looks good, pending verifying happy MIME types

**Pull request merged**